src/main/java/uk/gov/ons/ctp/response/sample/service/SampleSummaryActivationService.java

```diff
@@ -0,0 +1,135 @@
+package uk.gov.ons.ctp.response.sample.service;
+
+import static net.logstash.logback.argument.StructuredArguments.kv;
+
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.ons.ctp.response.sample.message.SampleSummaryActivationStatusPublisher;
+import uk.gov.ons.ctp.response.sample.representation.SampleSummaryActivationDTO;
+import uk.gov.ons.ctp.response.sample.representation.SampleSummaryStatusDTO;
+
+@Service
+public class SampleSummaryActivationService {
+
+  private static final Logger LOG = LoggerFactory.getLogger(SampleSummaryEnrichmentService.class);
+
+  @Autowired private SampleSummaryDistributionService sampleSummaryDistributionService;
+
+  @Autowired private SampleSummaryEnrichmentService sampleSummaryEnrichmentService;
+
+  @Autowired private SampleSummaryActivationStatusPublisher sampleSummaryActivationStatusPublisher;
+
+  /**
+   * Sorts out all the sample units for a given sample summary. It does this by first validating and
+   * enriching the data, followed by distributing all the sample units to case.
+   *
+   * <p>During each successful step of the process, a message will be sent to collection exercise so
+   * it can know how the sample summary activation is going.
+   *
+   * @param sampleSummaryActivation an object containing all the details required for activation
+   */
+  public void activateSampleSummaryFromPubsub(SampleSummaryActivationDTO sampleSummaryActivation)
+      throws SampleSummaryActivationException {
+    LOG.info(
+        "Beginning sample summary activation",
+        kv("sampleSummaryActivation", sampleSummaryActivation));
+    validateAndEnrich(sampleSummaryActivation);
+    sendEnrichStatusToCollectionExercise(sampleSummaryActivation.getCollectionExerciseId(), true);
+
+    distribute(sampleSummaryActivation);
+    sendDistributeStatusToCollectionExercise(
+        sampleSummaryActivation.getCollectionExerciseId(), true);
+    LOG.info(
+        "Completed sample summary activation",
+        kv("sampleSummaryActivation", sampleSummaryActivation));
```

Comment:
This kv command will do a toString call on the SampleSummaryActivationDTO object. Is that a readable format? Might be better to log out the collection exercise id instead?